src/sentry/ingest/billing_metrics_consumer.py

```diff
@@ -128,27 +128,28 @@ def __init__(
 
     def poll(self) -> None:
         while self._ongoing_billing_outcomes:
-            self._process_metrics_billing_bucket(timeout=None)
+            if not self._process_metrics_billing_bucket(timeout=None):
+                return
```

Comment:
nit: This can now be simplified to 

```python
while self._process_metrics_billing_bucket(timeout=None):
    pass
```    
(because that function also checks the length of `_ongoing_billing_outcomes`).

Reply:
Good suggestion! I had both conditions in the `while` and I found this easier, but yeah they weren't really needed.